src/region/regionUtil.ts

```diff
@@ -29,6 +44,98 @@ export const centerRegion = (shape: Shape): Position => {
     };
 };
 
+export const getPoints = (shape: Shape): [Point, Point, Point, Point] => {
+    const { height, width, x, y } = shape;
+
+    // Define the points from x,y and then in a clockwise fashion
+    // p1      p2
+    // +-------+
+    // |       |
+    // +-------+
+    // p4      p3
+    const p1 = { x, y };
+    const p2 = { x: x + width, y };
+    const p3 = { x: x + width, y: y + height };
+    const p4 = { x, y: y + height };
+
+    return [p1, p2, p3, p4];
+};
+
+export function selectTransformationPoint(shape: Shape, rotation: number): Point | null {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [p1, p2, p3, p4] = getPoints(shape);
+
+    // Determine which point will be the new x,y (as defined as the top left point) after rotation
+    // If -90deg: use p2
+    // If -180deg: use p3
+    // If -270deg: use p4
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return p2;
```

Comment:
Rather than calculating all four points each time, should we do the calculation inline here?

Reply:
The calculation isn't too complex, and it's more clear as to what's happening based on what `getPoints` does, imo